### ISSUE TITLE
Can not save and load a pyg model using torch.save

### ISSUE THREAD

**Issue by silver380** [pyg-team/pytorch_geometric#8104]
### 🐛 Describe the bug

Hi everyone. I've been trying to train a lightgcn model. I want to save a checkpoint for each 50 epoch. The problem is, if I use torch.save and then load the saved model, the loss of the model is the same as when I started to train it in the first epoch. Is there anything wrong with the following code?
model:
```
model = LightGCN(
    num_nodes=data.num_nodes,
    embedding_dim=64,
    num_layers=3,
).to(device)
optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
```

loading code:
```
checkpoint = torch.load('checkpoint_850.pt')
model.load_state_dict(checkpoint['model_state_dict'])
optimizer.load_state_dict(checkpoint['optimizer_state_dict'])
epoch = checkpoint['epoch']
model.train()
```

training and saving:
```
    history= {'loss' : []}
    save_interval = 50
    epochs = 1000

    def train(model, train_loader, device, lr, save_interval, epochs, start_epoch=1):
        #optimizer = torch.optim.Adam(model.parameters(), lr=lr)
        total_examples = len(train_loader.dataset)
        #progress_bar = tqdm(range(epochs), desc="Training", unit="epoch")

        for epoch in range(start_epoch, epochs+1):
            model.train()
            total_loss = 0
            total_example = 0
            for batch in tqdm(train_loader, colour='green'):
                optimizer.zero_grad()
                batch = batch.to(device)
                pos_rank, neg_rank = model(batch.edge_index, batch.edge_label_index).chunk(2)

                loss = model.recommendation_loss(
                pos_rank,
                neg_rank,
            ) #===========
                loss.backward()
                optimizer.step()
                total_loss += float(loss) * pos_rank.numel()
                total_examples += pos_rank.numel()

            avg_loss = total_loss / total_examples
            history['loss'].append(avg_loss)
        
            

            print(f"Epoch: {epoch}/{epochs}Loss: {avg_loss}")

            if epoch % save_interval == 0 and epoch != 0:
                torch.save({
                    'epoch': epoch,
                    'model_state_dict': model.state_dict(),
                    'optimizer_state_dict': optimizer.state_dict(),
                    }, f"checkpoint_{epoch}.pt")
                df = pd.DataFrame(history)
                df.to_csv('history.csv', index=False)
                print(f"Checkpoint{epoch} and history saved")

    train(model, train_loader, device, lr = 0.001, save_interval=save_interval, epochs=epochs, start_epoch=start_epoch)
```

### Environment

* PyG version:  2.3.1
* PyTorch version: 2.0.1+cu118
* OS: Windows 11
* Python version: 3.11.4
* CUDA/cuDNN version: 11.8

**Comment by rusty1s** [comment]
Mh, this is strange. Your code looks correct to me. Can you confirm that after loading, the model weights are indeed correctly set, e.g., by printing out `model.embedding.weight`?

**Comment by silver380** [comment]
I saved the model's final '''model.embedding.weight''' into a separate file, and after loading the model using my previous code, I also loaded the new file and checked to see if both of them were equal, and yes, it seems that it's loaded correctly and the model weights are correctly set.

**Comment by silver380** [comment]
It seems the problem is solved in PyG 2.4.
